api/handlers.go

```diff
@@ -1451,15 +1451,10 @@ func (si *ServerImplementation) fetchTransactions(ctx context.Context, filter id
 				return err
 			}
 
-			// Do not return inner transactions.
-			if tx.Id == nil {
-				continue
-			}
-
 			// The root txn has already been added.
 			// If we also want to return inner txns, we cannot deduplicate the
 			// results as inner txns all share the same txn ID as its root txn.
-			if _, ok := rootTxnDedupeMap[*tx.Id]; ok && !filter.ReturnInnerTxnOnly {
```

Comment:
I believe this extra check is no longer needed after modifying `txnRowToTransaction` to compute a txid from the inner transaction.